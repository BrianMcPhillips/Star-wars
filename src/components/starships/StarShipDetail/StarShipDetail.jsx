import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const StarShipDetail = ({
  name, 
  model,
  manufacturer,
  const_in_credits,
  length,
  max_atmosphering_speed,
  crew,
  passengers,
  hyperdrive_rating,
  starship_class,
  pilots,
  films
}) => (
  <div>
    <h1>{name}</h1>
    <h3>Model: {model}</h3>
    <h4>Manufacturer: {manufacturer}</h4>
    <h4>Cost: {const_in_credits} credits</h4>
    <h4>Length: {length}</h4>
    <h4>Max Speed: {max_atmosphering_speed} atmospheres</h4>
    <h5>No. of crew {crew}</h5>
    <h5>Passengers: {passengers}</h5>
    <h5>HyperDrive Rating: {hyperdrive_rating}</h5>
    <h5>Starship Class: {starship_class}</h5>
    <h4>Pilots: {pilots.map((pilot, i) =>
      <Link 
        to={`/char-detail/${pilot.split('people/')[1]}`}
        key={pilot + i}>
        <p>{pilot}</p>
      </Link>
    )}
    </h4>
    <h4>Films: {films.map((film, i) =>
      <Link 
        to={`/film-detail/${film.split('films/')[1]}`}
        key={film + i}>
        <p>{film}</p>
      </Link> 
    )}
    </h4>
  </div>
);

StarShipDetail.propTypes = {
  name: PropTypes.string.isRequired,
  model: PropTypes.string.isRequired,
  manufacturer: PropTypes.string.isRequired,
  const_in_credits: PropTypes.string.isRequired,
  length: PropTypes.string.isRequired,
  max_atmosphering_speed: PropTypes.string.isRequired,
  crew: PropTypes.string.isRequired,
  passengers: PropTypes.string.isRequired,
  hyperdrive_rating: PropTypes.string.isRequired,
  starship_class: PropTypes.string.isRequired,
  pilots: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
  films: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired
};

export default StarShipDetail;
