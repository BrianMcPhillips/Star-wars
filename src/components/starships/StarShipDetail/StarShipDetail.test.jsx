import React from 'react';
import { render, cleanup } from '@testing-library/react';
import StarShipDetail from './StarShipDetail';
import { MemoryRouter } from 'react-router-dom';

describe('StarShipDetail component', () => {
  afterEach(() => cleanup());
  it('renders StarShipDetail', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <StarShipDetail 
          name="X-wing" 
          model="T-65 X-wing" 
          manufacturer="Incom Corporation"
          const_in_credits="149999" 
          length="12.5"
          max_atmosphering_speed="1050"
          crew="1"
          passengers="0"
          hyperdrive_rating="1.0"
          starship_class="Starfighter"
          pilots= {[
            'https://swapi.dev/api/people/1/', 
            'https://swapi.dev/api/people/2/', 
            'https://swapi.dev/api/people/3/', 
            'https://swapi.dev/api/people/4/', 
            'https://swapi.dev/api/people/5/'
          ]}
          films= {[
            'https://swapi.dev/api/films/1/', 
            'https://swapi.dev/api/films/2/', 
            'https://swapi.dev/api/films/3/'
          ]} 
        />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
