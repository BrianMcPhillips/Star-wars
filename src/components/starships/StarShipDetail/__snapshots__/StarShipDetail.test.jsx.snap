// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StarShipDetail component renders StarShipDetail 1`] = `
<DocumentFragment>
  <div>
    <h1>
      X-wing
    </h1>
    <h3>
      Model: T-65 X-wing
    </h3>
    <h4>
      Manufacturer: Incom Corporation
    </h4>
    <h4>
      Cost: 149999 credits
    </h4>
    <h4>
      Length: 12.5
    </h4>
    <h4>
      Max Speed: 1050 atmospheres
    </h4>
    <h5>
      No. of crew: 1
    </h5>
    <h5>
      Passengers: 0
    </h5>
    <h5>
      HyperDrive Rating: 1.0
    </h5>
    <h5>
      Starship Class: Starfighter
    </h5>
    <h4>
      Pilots: 
      <a
        href="/char-detail/1/"
      >
        <p>
          https://swapi.dev/api/people/1/
        </p>
      </a>
      <a
        href="/char-detail/2/"
      >
        <p>
          https://swapi.dev/api/people/2/
        </p>
      </a>
      <a
        href="/char-detail/3/"
      >
        <p>
          https://swapi.dev/api/people/3/
        </p>
      </a>
      <a
        href="/char-detail/4/"
      >
        <p>
          https://swapi.dev/api/people/4/
        </p>
      </a>
      <a
        href="/char-detail/5/"
      >
        <p>
          https://swapi.dev/api/people/5/
        </p>
      </a>
    </h4>
    <h4>
      Films: 
      <a
        href="/film-detail/1/"
      >
        <p>
          https://swapi.dev/api/films/1/
        </p>
      </a>
      <a
        href="/film-detail/2/"
      >
        <p>
          https://swapi.dev/api/films/2/
        </p>
      </a>
      <a
        href="/film-detail/3/"
      >
        <p>
          https://swapi.dev/api/films/3/
        </p>
      </a>
    </h4>
  </div>
</DocumentFragment>
`;
